ts/modules/export-data.src.ts

```diff
@@ -1162,25 +1171,46 @@ Chart.prototype.downloadXLS = function (): void {
  * @fires Highcharts.Chart#event:afterViewData
  */
 Chart.prototype.viewData = function (): void {
```

Comment:
I don't think we should modify `viewData()`. This is our API method, which, well, should display the table. Now it's toggling the data, so how about this (pseudo code):

```js
Chart.toggleDataTable = function () {
  if (table) {
    this.hideData();
  } else {
    this.showData();
  }
}
```

Then, we link `toggleDataTable` to exporting button and we don't have to worry further. Plus we make equivalent API method for `showData` we already have, plus short method for toggling (can also be in API).

Could you also check why `event:afterViewData` was added? Perhaps we should already implement `afterHideData` event.

Reply:
The `event:afterViewData` was added to focus on the data table when it appears.
But here I'm not sure how this should behave after hiding the data table.
@oysteinmoseng , might you give us an opinion about it?

Reply:
When we hide the table we would not want to change the focus. So as long as the event is fired after the table is shown we are good, I think